src/items.rs

```diff
@@ -1165,10 +1165,10 @@ pub(crate) fn format_trait(
     result.push_str(&generics_str);
 
     // FIXME(#2055): rustfmt fails to format when there are comments between trait bounds.
+    // Retrieve *unnormalized* ident (See #6069)
+    let source_ident = context.snippet(item.ident.span);
     if !bounds.is_empty() {
-        let ident_hi = context
-            .snippet_provider
-            .span_after(item.span, item.ident.as_str());
+        let ident_hi = context.snippet_provider.span_after(item.span, source_ident);
```

Comment:
nit: Can you define `source_ident` in the `if-block`. I think it's a little confusing that the unnormalized ident comment follows the FIXME. I want to avoid any confusion with the adjacent comments and define variables closer to where they're being used.